packages/aws-cdk-lib/aws-dynamodb/test/dynamodb.test.ts

```diff
@@ -445,6 +445,77 @@ testDeprecated('when specifying every property', () => {
   );
 });
 
+test('when replica removal policy is not specified', () => {
+  const stack = new Stack();
+  new Table(stack, CONSTRUCT_NAME, {
+    tableName: TABLE_NAME,
+    partitionKey: TABLE_PARTITION_KEY,
+    removalPolicy: RemovalPolicy.RETAIN,
+    replicationRegions: ['eu-west-2', 'eu-west-3'],
+  });
+
+  Template.fromStack(stack).hasResourceProperties('Custom::DynamoDBReplica', {
+    'SkipReplicaDeletion': true,
+  });
+});
+
+test('when replica removal policy is not specified', () => {
+  const stack = new Stack();
+  const table = new Table(stack, CONSTRUCT_NAME, {
+    tableName: TABLE_NAME,
+    partitionKey: TABLE_PARTITION_KEY,
+    replicationRegions: ['eu-west-2', 'eu-west-3'],
+  });
+  table.applyRemovalPolicy(RemovalPolicy.DESTROY);
+
+  Template.fromStack(stack).hasResourceProperties('Custom::DynamoDBReplica', {
+    'SkipReplicaDeletion': false,
+  });
+});
+
+test('when replica and table removal policy is not specified', () => {
```

Comment:
I assume before this PR, when replica and table removal policy is not specified, removing the resource will always delete the replica. So this is kind of a default behavior change. Do we consider this a breaking change?

Reply:
Good question! I thought about this and decided that this is not a breaking change since 
1. Table policy is default to retain by default, replica should be retain by default as well. The current behaviour is bug and potentially cause customer downtime in replica regions.
2. Think about global table behaviour that replica is always linked to a source table. Global table should not delete replica when table is removed from stack.
3. This change is not a breaking change because persisting a table replica will not cause replacement or downtime or any other potential problems that I can think of.

But I can also add a feature flag for safety purpose to avoid any unconsidered cases

Reply:
Got it. Yeah that makes sense to me. The change shouldn't cause downtime or something similar to users.